pkg/controllers/management/auth/globalroles/enqueue.go

```diff
@@ -239,3 +242,83 @@ func (g *globalRBACEnqueuer) namespaceEnqueueGR(_, _ string, obj runtime.Object)
 	}
 	return grNames, nil
 }
+
+// fleetWorkspaceEnqueueGRB enqueues GlobalRole that have set InheritedFleetWorkspacePermissions
+// when a FleetWorkspace has changed.
+func (g *globalRBACEnqueuer) fleetWorkspaceEnqueueGR(_, _ string, obj runtime.Object) ([]relatedresource.Key, error) {
```

Comment:
For the `inheritedClusterRoles` part, we add an annotation so that future calls to this method won't cause a mass-re-enqueue of all matching global roles. You can see that [here](https://github.com/rancher/rancher/blob/6148fd36c961f3d3e3210478b686348d7f6f994f/pkg/controllers/management/auth/globalroles/enqueue.go#L117). While I think that this is important for the cluster use case (since I expect that object to see a lot of changes), I don't think it's strictly as important for this use case. Can you take a look at this and see if you want to add something similar here?

Reply:
I don't think we need that in this case. `FleetWorkspace`  just contains `metav1.ObjectMeta`, and we just use the name of the workspace to generate the backing namespace. Unlike `Clusters`, no field in a `FleetWorkspace` is expected to change that would trigger a re-enqueue of all matching `GlobalRoles`.

Reply:
Alright, that's ok. Let's keep an eye on this from a performance standpoint, but we don't need to fix this here.